certbot-dns-google/certbot_dns_google/__init__.py

```diff
@@ -25,45 +32,36 @@
+DNS managed zone(s) for which certificates are being issued.
 
-* ``dns.changes.create``
-* ``dns.changes.get``
-* ``dns.changes.list``
-* ``dns.managedZones.get``
-* ``dns.managedZones.list``
-* ``dns.resourceRecordSets.create``
-* ``dns.resourceRecordSets.delete``
-* ``dns.resourceRecordSets.list``
-* ``dns.resourceRecordSets.update``
+In most cases, configuring credentials for Certbot will require `creating a service account
+<https://cloud.google.com/iam/docs/service-accounts-create>`_ and granting the `required
+permissions`_ using IAM.
 
-(The closest role is `dns.admin <https://cloud.google.com/dns/docs/
-access-control#dns.admin>`_).
 
-If the above permissions are assigned at the `resource level <https://cloud
-.google.com/dns/docs/zones/iam-per-resource-zones>`_, the same user must
-have, at the PROJECT level, the following permissions:
+Providing Credentials
+^^^^^^^^^^^^^^^^^^^^^
 
-* ``dns.managedZones.get``
-* ``dns.managedZones.list``
+The preferred method of providing credentials is to `set up Application Default Credentials
+<https://cloud.google.com/docs/authentication/provide-credentials-adc>`_ (ADC) in the environment
+that Certbot is running in.
 
-(The closest role is `dns.reader <https://cloud.google.com/dns/docs/
-access-control#dns.reader>`_).
+If you are running Certbot on Google Cloud then a service account can be assigned directly to most
+types of workload, including `Compute Engine VMs <https://cloud.google.com/compute/docs/access/
+create-enable-service-accounts-for-instances>`_, `Kubernetes Engine Pods <https://cloud.google.com/
+kubernetes-engine/docs/how-to/workload-identity>`_, `Cloud Run jobs <https://cloud.google.com/run
+/docs/securing/service-identity>`_, `Cloud Functions <https://cloud.google.com/functions/docs/
+securing/function-identity>`_, and `Cloud Builds <https://cloud.google.com/build/docs/securing-
+builds/configure-user-specified-service-accounts>`_.
 
-Google provides instructions for `creating a service account <https://developers
-.google.com/identity/protocols/OAuth2ServiceAccount#creatinganaccount>`_ and
-`information about the required permissions <https://cloud.google.com/dns/access
--control#permissions_and_roles>`_. If you're running on Google Compute Engine,
-you can `assign the service account to the instance <https://cloud.google.com/
-compute/docs/access/create-enable-service-accounts-for-instances>`_ which
-is running certbot. A credentials file is not required in this case, as they
-are automatically obtained by certbot through the `metadata service
-<https://cloud.google.com/compute/docs/storing-retrieving-metadata>`_ .
+If you are not running Certbot on Google Cloud then a credentials file should be generated using one
+of the methods outlined in the `steps for authenticating from on-premises or another cloud provider
+<https://cloud.google.com/docs/authentication/provide-credentials-adc#on-prem>`_.
```

Comment:
I see where you're coming from. I've tweaked that section to make passing service account keys to `--dns-google-credentials` the first thing seen. I've also added a small bit about WIF at the very end, but I'm happy to omit that if you think it would cause undue confusion.